### PR TITLE
fix: invalid aria-label

### PR DESCRIPTION
**Pull request by paul-balchin-ibm**
Contributes to #2978

Issue: 
- assigning a node to aria-label like `aria-label={<div>Hello <b>there</b>.</span>}` in React, 
- renders `aria-label="[object Object]"` in HTML.

#### What did you change?
- new function converts node text to string, `aria-label={getNodeTextContent(the same node)}`,
- renders `aria-label="Hello there."`.

#### How did you test and verify your work?

- Tested locally.
- Included a test file `getNodeTextContent.test.js`.

### CODE REVIEW

**Comment by paul-balchin-ibm** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/ibm-products/src/global/js/utils/getNodeTextContent.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+//
+// Copyright IBM Corp. 2021, 2021
```
Comment (elycheea) [inline packages/ibm-products/src/global/js/utils/getNodeTextContent.test.js:2]:
```suggestion
// Copyright IBM Corp. 2023, 2023
```

**Review by elycheea — COMMENTED** [review]
Ooh, interesting approach that does allow us not to add a potentially duplicative prop. 🤔 Separate `aria-label` prop could come up again at some point, but kind of like this approach too. Good to go other than fixing some copyright dates. :P

**Comment by paul-balchin-ibm** [comment]
image

**Comment by paul-balchin-ibm** [comment]
> Separate `aria-label` prop could come up again at some point...

I'm pretty sure there are. I've also seen other console warnings/errors, when running Storybook as `localhost` such as these just for http://localhost:3000/?path=/story/ibm-products-components-datagrid-datagrid-canary--empty-state
- "Received `false` for a non-boolean attribute `role`"
- "React does not recognize the `renderIcon` prop on a DOM element"
- "React does not recognize the `placeHolderText` prop on a DOM element"
- "Invalid argument supplied to oneOfType, expected an instance of array"
- "Invalid arguments supplied to oneOf, expected an array, got 4 arguments"

I think it would be a good exercise to go through the stories and quash these issues. As a an app dev, I don't want to have filter my own errors from those generated by a library's components.

**Review by elycheea — APPROVED** [review]

**Comment by wkeese** [comment]
Hi @paul-balchin-ibm - You've fixed this situation:

```
<Tearsheet
title={<div>bill was here</div>}
```

but when there's a level of indirection it doesn't work:

```
  const Title = () => <div>bill was here</div>;
...
<Tearsheet
title={<Title/>}
```

In that case there's no aria-label for the role=dialog node at all.

**Comment by wkeese** [comment]
I've filed https://github.com/carbon-design-system/ibm-products/issues/4013, which addresses having a separate aria-label prop.  That seems like the final solution to this problem.

**Pull request merged**